File: app/src/main/java/databinding/example/com/java/ui/ObservableFieldActivity.java
package databinding.example.com.java.ui;

import android.databinding.DataBindingUtil;
import android.databinding.ObservableInt;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import databinding.example.com.R;
import databinding.example.com.databinding.JavaObservableFieldProfileBinding;
import databinding.example.com.java.data.ObservableFieldProfile;

public class ObservableFieldActivity extends AppCompatActivity {
    private ObservableFieldProfile observableFieldProfile = new ObservableFieldProfile("java over", "hi", new ObservableInt(0));

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        JavaObservableFieldProfileBinding binding =
                DataBindingUtil.setContentView(this, R.layout.java_observable_field_profile);
        binding.setUser(observableFieldProfile);


    }

    public void onLike(View view) {
        observableFieldProfile.likes.set(observableFieldProfile.likes.get() + 1);
    }
}

File: app/src/main/java/databinding/example/com/basic/ui/ObservableFieldActivity.kt
package databinding.example.com.basic.ui

import android.databinding.DataBindingUtil
import android.databinding.ObservableInt
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import databinding.example.com.R
import databinding.example.com.basic.data.ObservableFieldProfile
import databinding.example.com.databinding.ObservableFieldProfileBinding

class ObservableFieldActivity : AppCompatActivity() {

    private val observableFieldProfile = ObservableFieldProfile("Ada11", "Lovelace", ObservableInt(0))

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val binding: ObservableFieldProfileBinding =
                DataBindingUtil.setContentView(this, R.layout.observable_field_profile)
        binding.user = observableFieldProfile
    }

    fun onLike(view: View) {
        observableFieldProfile.likes.set(observableFieldProfile.likes.get() + 1)
    }
}